web/src/components/tasks/TaxTask.test.tsx

```diff
@@ -204,4 +205,54 @@ describe("<TaxTask />", () => {
       });
     });
   });
+
+  describe("tax id disclaimer", () => {
+    const renderComponent = (initialUserData?: UserData) => {
+      render(
+        <WithStatefulUserData initialUserData={initialUserData ?? generateUserData({})}>
+          <TaxTask task={task} />
+        </WithStatefulUserData>
+      );
+    };
+
+    it("shows disclaimer for trade name legal structure", () => {
+      const userData = generateUserData({
+        profileData: generateProfileData({
+          legalStructureId: randomTradeNameLegalStructure(),
+        }),
+      });
+      renderComponent(userData);
+
+      expect(screen.getByTestId("tax-disclaimer")).toHaveTextContent(
+        markdownToText(Config.profileDefaults.fields.taxId.default.disclaimerMd)
+      );
+    });
+
+    it("does not show disclaimer for public filing legal structure", () => {
+      const userData = generateUserData({
+        profileData: generateProfileData({
+          legalStructureId: randomPublicFilingLegalStructure(),
+        }),
+      });
+      renderComponent(userData);
+
+      expect(screen.queryByTestId("tax-disclaimer")).not.toBeInTheDocument();
+    });
+  });
 });
+
+const randomTradeNameLegalStructure = () => {
+  const tradeNameLegalStructures = LegalStructures.filter((x) => {
+    return x.hasTradeName;
+  });
+  const randomIndex = Math.floor(Math.random() * tradeNameLegalStructures.length);
+  return tradeNameLegalStructures[randomIndex].id;
+};
+
+const randomPublicFilingLegalStructure = () => {
+  const nonTradeNameLegalStructures = LegalStructures.filter((x) => {
+    return x.requiresPublicFiling;
+  });
+  const randomIndex = Math.floor(Math.random() * nonTradeNameLegalStructures.length);
+  return nonTradeNameLegalStructures[randomIndex].id;
+};
```

Comment:
I think these functions already exist in helpers/support, we shouldn't need to recreate them